src/components/ec-file-upload/ec-file-upload.vue

```diff
@@ -42,47 +42,42 @@
   </div>
 </template>
 
-<script>
+<script setup>
 import EcFileDropzone from '../ec-file-dropzone';
 import EcFileList from '../ec-file-list';
 
-export default {
-  name: 'EcFileUpload',
-  components: {
-    EcFileDropzone,
-    EcFileList,
+const emit = defineEmits(['update:modelValue', 'change']);
+
+const props = defineProps({
+  modelValue: {
+    type: Array,
+    default: () => ([]),
+  },
+  label: {
+    type: String,
   },
-  props: {
-    modelValue: {
-      type: Array,
-      default: () => ([]),
-    },
-    label: {
-      type: String,
-    },
-    note: {
-      type: String,
-    },
-    isDisabled: {
-      type: Boolean,
-    },
+  note: {
+    type: String,
   },
-  emits: ['update:modelValue', 'change'],
-  methods: {
-    onChange(newFiles) {
-      const newFilesNames = new Set(newFiles.map(file => file.name));
-      const updatedFileList = this.modelValue.filter(prevFile => !newFilesNames.has(prevFile.name));
-      this.update([...updatedFileList, ...newFiles]);
-    },
-    onDelete(fileToDelete) {
-      this.update(this.modelValue.filter(fileItem => fileItem !== fileToDelete));
-    },
-    update(items) {
-      this.$emit('update:modelValue', items);
-      this.$emit('change', items);
-    },
+  isDisabled: {
+    type: Boolean,
   },
-};
+});
+
+function onChange(newFiles) {
+  const newFilesNames = new Set(newFiles.map(file => file.name));
+  const updatedFileList = props.modelValue.filter(prevFile => !newFilesNames.has(prevFile.name));
+  update([...updatedFileList, ...newFiles]);
+}
+
+function onDelete(fileToDelete) {
+  update(props.modelValue.filter(fileItem => fileItem !== fileToDelete));
+}
+
+function update(items) {
+  emit('update:modelValue', items);
+  emit('change', items);
```

Comment:
Why we emit here two identic events? We could stick with one `change`. I would understand if we emitted here updated items, I mean **new** files in `update:modelValue` event, but we don't.

Reply:
The `change` emit is filtered upwards into EBO and is used to trigger file validation. This can in the future be refactored into the watch function in those two files, and then removed here.

Reply:
I propose to stay with `change` and remove `update:modelValue`. BTW, I didn't find using of update:modelValue in EBO.

Reply:
The `update:modelValue` updates the v-model of the component which contains the file(s) from what I can see so maybe refactoring the `change` event in the future could be the better route.

Reply:
OMG, native update:modelValue event, I see :).